musicround/helpers/catalog_backfill.py

```
@@ -0,0 +1,233 @@
+                if not dry_run:
+                    for sid in newly_mapped_spotify_ids:
+                        song = Song.query.filter_by(spotify_id=sid).first()
+                        if song:
+                            ImportHelper._fetch_audio_features_for_song(sp, song, sid)
+                            if song.danceability is not None:
+                                updated_c += 1
+                            if sleep_sec > 0:
+                                time.sleep(sleep_sec)
+                    db.session.commit()
+                else:
+                    for sid in newly_mapped_spotify_ids:
```

Ruff (0.15.20)

[warning] Loop control variable `sid` not used within loop body

(B007)